contracts/base/Dispatcher.sol

```diff
@@ -197,6 +209,27 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks
         if (!success) ERC721(token).transferFrom(address(this), returnAddress, id);
     }
 
+    function approveAndSell1155(bytes memory inputs, address protocol)
+        internal
+        returns (bool success, bytes memory output)
+    {
+        (bytes memory data, address token, address spender, uint256 id, uint256 amount, address returnAddress) =
+            abi.decode(inputs, (bytes, address, address, uint256, uint256, address));
+
+        ERC1155(token).setApprovalForAll(spender, true);
+
+        (success, output) = protocol.call(data);
+
+        // Revoke approval
+        // TODO: is this correct?
+        ERC1155(token).setApprovalForAll(spender, false);
```

Comment:
I _think_ we can just leave the approval, given the router is never meant to hold standing balance of anything

Reply:
Alright, not sure how much gas overhead this is too

Reply:
double checking w/ @hensha256 @ewilz

Reply:
as long as we're always calling Permit2 to transfer tokens from the msg.sender, I think it should be fine to have "hanging approvals". I was trying hard a few months ago to try and think of vulnerabilities for hanging approvals, but couldn't think of any, but I do think we should give some extra scrutiny to the topic and make a full fledged decision that applies to all token types eventually. Fine to have hanging approvals for now as a default...no sure what @hensha256 thinks

Reply:
yeah i cant really think of many issues from hanging approvals so should be okay. Though once its been approved once it then is a waste of user for other users to have line 220 at all. So maybe we should have a separate approve721 command, that the first user to sell an NFT includes in their tx?

Reply:
Do you mean approve1155? And that once the router approves Seaport for the entire 1155 collection we shouldn't have another approval? If we want to do hanging approvals we could do something like 
```solidity
if(!isApprovedForAll(address(this), spender) {
 ERC1155(token).setApprovalForAll(spender, true);
}
```

Reply:
@hensha256 means that `approve` and `sell` are 2 completely different commands. You could actually imagine collection creators triggering this to save their traders gas. Interesting idea, and a nice-to-have depending on gas savings.

Reply:
Ah I see, yeah that makes sense to me. I'd support decoupling the approve from the selling command - however this would only make sense for 1155 right? Since for 721 approval is cleared on transfer